src/panels/config/zha/ha-config-zha.ts

```diff
@@ -1,26 +1,41 @@
-import { html, LitElement, PropertyDeclarations } from "@polymer/lit-element";
-import { TemplateResult } from "lit-html";
 import "@polymer/app-layout/app-header/app-header";
 import "@polymer/app-layout/app-toolbar/app-toolbar";
-import "@polymer/paper-icon-button/paper-icon-button";
 import "@polymer/iron-flex-layout/iron-flex-layout-classes";
-import { HomeAssistant } from "../../../types";
-
+import { html, LitElement, PropertyDeclarations } from "@polymer/lit-element";
+import "@polymer/paper-icon-button/paper-icon-button";
+import { HassEntity } from "home-assistant-js-websocket";
+import { TemplateResult } from "lit-html";
+import { HASSDomEvent } from "../../../common/dom/fire_event";
+import { Cluster } from "../../../data/zha";
 import "../../../layouts/ha-app-layout";
 import "../../../resources/ha-style";
-
+import { HomeAssistant } from "../../../types";
+import {
+  ZHAClusterSelectedParams,
+  ZHAEntitySelectedParams,
+  ZHANodeSelectedParams,
+} from "./types";
+import "./zha-cluster-attributes";
+import "./zha-cluster-commands";
 import "./zha-network";
+import "./zha-node";
 
 export class HaConfigZha extends LitElement {
   public hass?: HomeAssistant;
   public isWide?: boolean;
   private _haStyle?: DocumentFragment;
   private _ironFlex?: DocumentFragment;
+  private _selectedNode?: HassEntity;
```

Comment:
Do all nodes map to an entity in HA ? 

Are these 3 selected things mutually exclusive or can they co-exist?

Reply:
This is a nuanced answer... maybe better suited for discord. Short answer: node is essentially device. It can have 1 or many entities. Each entity can have 1 or many clusters. each cluster can have 1 or many commands and 1 or many attributes.

Reply:
Okay, what about a remote, that won't have an entity but just generate events, no?

Reply:
Would it make sense to just fetch all nodes via the websocket command and use that ? Because now you might end up adding more things to entities to satisfy a config panel.

Reply:
And I also suggest that you make a selection type that contains all three selections.

```
type ZHASelection {
  node: string;
  cluster: string;
}
```

Reply:
the selections are mainly propagated to clear selections that aren't valid when another is changed... does that make sense? Everything is built off of drilling down. 

Ex: I Pick a node -> pick an entity -> pick a cluster -> pick an attribute. 

If I subsequently change the entity I have selected I then have to clear the selected cluster and attribute that I had already selected because the selections aren't valid and the data in each picker changes. 

I am using events to propagate information up the dom and bindings to propagate data down the dom so that independent components can interact. Did I do something wrong here?

I don't see what ZHASelection gives me here esp since the components that fire events may not have all of the parts to populate it correctly. Am I making sense?